lib/irb.rb

```diff
@@ -506,13 +506,14 @@ def eval_input
 
       @scanner.set_auto_indent(@context) if @context.auto_indent_mode
 
-      @scanner.each_top_level_statement do |line, line_no|
+      @scanner.each_top_level_statement(@context) do |line, line_no|
         signal_status(:IN_EVAL) do
           begin
             line.untaint if RUBY_VERSION < '2.7'
             if IRB.conf[:MEASURE] && IRB.conf[:MEASURE_CALLBACKS].empty?
               IRB.set_measure_callback
             end
+            is_assignment = assignment_expression?(line)
```

Comment:
Why moving this line here though? It doesn't look like to change the behavior?

Reply:
It actually changes the behavior in a rare case.
```ruby
irb = IRB::Irb.new
context = irb.instance_variable_get('@context')

code = "a /'/i if false; a=1; x=1000.times.to_a#'.size"
irb.assignment_expression?(code) #=> true
context.evaluate(code, 1) #=> [0,1,2,3,...] assignment to local variable x
irb.assignment_expression?(code) #=> false
context.evaluate(code, 1) #=> 0 not an assignment expression
```

`assignment_expression?` check should be done before `@context.evaluate` (in L519 and L530)

I added test case for this

Reply:
Wow thanks for spotting this. I think it also worth a comment then?

Reply:
Thank you for the example 👍